libpod/image/image.go

```diff
@@ -335,6 +335,16 @@ func (i *Image) Names() []string {
 	return i.image.Names
 }
 
+// NamesHistory returns a string array of names previously associated with the
+// image, which may be a mixture of tags and digests
+func (i *Image) NamesHistory() []string {
+	if len(i.image.Names) > 0 && len(i.image.NamesHistory) > 0 &&
+		i.image.Names[len(i.image.Names)-1] == i.image.NamesHistory[0] {
```

Comment:
`i.image.Names` gets appended on re-tag, which somehow overlaps with the `NamesHistory` feature.

Reply:
Can you add a comment to this condition? It's sufficiently complex that I will have forgotten the meaning by this evening :^)

Reply:
Sure, I'll add a note. :+1: